WordPress/Classes/ViewRelated/Post/WPPostViewController.m

```diff
@@ -1850,48 +1915,21 @@ - (void)addDeviceMediaAsset:(PHAsset *)asset
 {
     MediaService *mediaService = [[MediaService alloc] initWithManagedObjectContext:[[ContextManager sharedInstance] mainContext]];
     __weak __typeof__(self) weakSelf = self;
+    bool isImage = (asset.mediaType == PHAssetMediaTypeImage);
     NSString *mediaUniqueID = [self uniqueIdForMedia];
     [mediaService createMediaWithPHAsset:asset
                          forPostObjectID:self.post.objectID
                        thumbnailCallback:^(NSURL *thumbnailURL) {
                            __typeof__(self) strongSelf = weakSelf;
-                           if (!strongSelf) {
-                               return;
+                           if (strongSelf) {
+                               [strongSelf createMediaThumbnailCallback:thumbnailURL mediaID:mediaUniqueID isImage:isImage];
                            }
-                           [[NSOperationQueue mainQueue] addOperationWithBlock:^{
```

Comment:
Nice going pulling out these sections so they're not duplicated for the media and document pickers 👍